distsql/select_result.go

```diff
@@ -167,6 +173,27 @@ func (r *selectResult) getSelectResp() error {
 	}
 }
 
+func (r *selectResult) updateCopRuntimeStats(callee string) {
+	if r.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl == nil || callee == "" {
+		return
+	}
+	if len(r.selectResp.GetExecutionSummaries()) != len(r.copPlanIDs) {
+		log.Errorf("invalid cop task execution summaries length, expected: %v, received: %v",
+			len(r.copPlanIDs), len(r.selectResp.GetExecutionSummaries()))
+		return
+	}
+
+	for i, detail := range r.selectResp.GetExecutionSummaries() {
+		if detail != nil && detail.TimeProcessedNs != nil &&
+			detail.NumProducedRows != nil && detail.NumIterations != nil {
```

Comment:
What if the 3 attributes are nil? It seems that some info will be lost, I think this is not acceptable?

Reply:
emm.. Is there anything we can do more? The only thing I can think of is printing some logs.. Do you have any suggestion?